compiler/noirc_frontend/src/graph/mod.rs

```diff
@@ -90,6 +91,14 @@ mod crate_name {
             assert!(!CrateName::is_valid_name(&bad_char_string));
         }
     }
+
+    #[test]
+    fn it_can_serialize_and_deserialize() {
+        let crate_name = "test_crate".parse::<CrateName>().unwrap();
+        let serialized = serde_json::to_string(&crate_name).unwrap();
+        let deserialized = serde_json::from_str::<CrateName>(&serialized).unwrap();
+        assert_eq!(crate_name, deserialized);
+    }
```

Comment:
:facepalm: you're right. This code is effectively testing serde. How about something like

```
#[test]
#[should_panic]
fn rejects_bad_names() {
  serde_json::from_str::<CrateName>("bad-name").unwrap();
}
```

Reply:
Something like that but we shouldn't panic, we should just return an error.

Reply:
Ah, I see this is what you were doing with the unwrap. That's a little implicit. I think it would be better to directly assert on the return type being an error (we can then check the error type to be what we expect).

Reply:
Changed it with a `assert!(parse.is_err())` test :)